packages/lib/services/plugins/api/JoplinSettings.ts

```diff
@@ -39,32 +39,48 @@ export default class JoplinSettings {
 	}
 
 	/**
-	 * Registers a new setting. Note that registering a setting item is dynamic and will be gone next time Joplin starts.
+	 * Registers new settings.
+	 * Note that registering a setting item is dynamic and will be gone next time Joplin starts.
 	 * What it means is that you need to register the setting every time the plugin starts (for example in the onStart event).
 	 * The setting value however will be preserved from one launch to the next so there is no risk that it will be lost even if for some
 	 * reason the plugin fails to start at some point.
 	 */
+	public async registerSettings(settings: Record<string, SettingItem>) {
+		const settingsPromises = [];
+
+		for (const [key, setting] of Object.entries(settings)) {
+			const internalSettingItem: InternalSettingItem = {
+				key: key,
+				value: setting.value,
+				type: setting.type,
+				public: setting.public,
+				label: () => setting.label,
+				description: (_appType: string) => setting.description,
+			};
+
+			if ('isEnum' in setting) internalSettingItem.isEnum = setting.isEnum;
+			if ('section' in setting) internalSettingItem.section = this.namespacedKey(setting.section);
+			if ('options' in setting) internalSettingItem.options = () => setting.options;
+			if ('appTypes' in setting) internalSettingItem.appTypes = setting.appTypes;
+			if ('secure' in setting) internalSettingItem.secure = setting.secure;
+			if ('advanced' in setting) internalSettingItem.advanced = setting.advanced;
+			if ('minimum' in setting) internalSettingItem.minimum = setting.minimum;
+			if ('maximum' in setting) internalSettingItem.maximum = setting.maximum;
+			if ('step' in setting) internalSettingItem.step = setting.step;
+
+			settingsPromises.push(Setting.registerSetting(this.namespacedKey(key), internalSettingItem));
+		}
+		return Promise.all(settingsPromises);
```

Comment:
Weren't you going to get rid of Promise.all?

Reply:
the standard deviation of the code using Promise.all is lower (https://gist.github.com/jalajcodes/81138432b1f875467fef2565b96fa758)
Isn't that better?